src/kernel.ts

```diff
@@ -422,6 +459,9 @@ class Kernel implements IKernel {
    * received and validated.
    */
   kernelInfo(): Promise<IKernelInfo> {
+    if (this._status === KernelStatus.Dead) {
+      return Promise.reject(Error('Kernel is not ready to send a message'));
+    }
```

Comment:
Shouldn't these status checks be in sendKernelMessage (like you already have done in sendShellMessage) to avoid the duplication?